Sources/SwiftInspectorCore/Tests/InitializerAnalyzerSpec.swift

```diff
@@ -0,0 +1,135 @@
+// Created by Francisco Diaz on 3/27/20.
+//
+// Copyright (c) 2020 Francisco Diaz
+//
+// Distributed under the MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import Nimble
+import Quick
+import Foundation
+
+@testable import SwiftInspectorCore
+
+final class InitializerAnalyzerSpec: QuickSpec {
```

Comment:
What is supposed to happen when the same type exists in multiple places in a file?

```swift
class Foo {
  init(myParam: String) { }
}

struct Bar {
  class Foo {
    init(anotherParam: Int) { }
  }
}
```

Reply:
We should be returning both initializers. Let me add a test for that.

Reply:
Right now we'd be only returning the top-level one actually. I'm not sure what's more desirable. Maybe we should create an option to look for the scoped ones? Let's discuss and add this in another PR before we ship a new version of this.

Reply:
https://github.com/fdiaz/SwiftInspector/issues/18